src/index.js

```diff
@@ -209,20 +209,18 @@ class Router extends Component {
 	}
 
 	getMatchingChildren(children, url, invoke) {
-		return children.slice().sort(pathRankSort).map( vnode => {
-			let attrs = vnode.attributes || {},
-				path = attrs.path,
-				matches = exec(url, path, attrs);
-			if (matches) {
-				if (invoke!==false) {
-					let newProps = { url, matches };
-					assign(newProps, matches);
-					return cloneElement(vnode, newProps);
-				}
-				return vnode;
-			}
-			return false;
-		}).filter(Boolean);
+		const routes = getMatchingRoutes(children, url);
+
+		if (invoke!==false){
+			return routes.map(r => {
+				let { route, matches } = r;
```

Comment:
Thought:

```js
getMatchingChildren(children, url, invoke) {
	return getMatchingRoutes(children, url)
		.map( ({ route, matches }) => {
			if (invoke!==false) {
				let newProps = { matches, url };
				assign(newProps, matches);
				// note: there's a merge conflict here that adds the following 2 lines:
				delete newProps.key;
				delete newProps.ref;
				route = cloneElement(route, newProps);
			}
			return route;
		});
}
```

Reply:
After rebase, current size of the lib on your repository:

```
$ npm run size

> preact-router@2.5.7 size ~/Desktop/github/preact-router
> size=$(gzip-size $npm_package_main) && echo "gzip size: $size / $(pretty-bytes $size)"

gzip size: 1991 / 1.99 kB
```

With the suggested refactoring it changes to **gzip size: 1991 / 1.99 kB**. We save one glorious byte. Do you still feel it is worth to have the conditional inside the map method? If that's the case I'm going to update the PR as soon as possible*, but please give me a confirm.

* I usually prefer code readability over byte savings, but of course it's your choice... and thanks for your efforts btw.